tests/test_cli.py

```diff
@@ -0,0 +1,151 @@
+    registry = base_app.extensions['invenio-records-resources'].registry
+    registry.register(names_service, service_id='rdm-names')
+
+    yield base_app
+
+
+@pytest.fixture(scope='module')
+def app_config(app_config):
+    """Mimic an instance's configuration."""
+    app_config["VOCABULARIES_DATASTREAM_TRANSFORMERS"] = {
+        "orcid-xml": OrcidXMLTransformer
+    }
+    app_config["VOCABULARIES_DATASTREAM_WRITERS"] = {
+        "names-service": NamesServiceWriter
+    }
+
+    return app_config
+
+
+@pytest.fixture(scope='module')
+def name_xml():
+    return (
+        '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>\n'
+        '<record:record path="/0000-0001-8135-3489">\n'
+        '    <common:orcid-identifier>\n'
+        '        <common:uri>https://orcid.org/0000-0001-8135-3489</common:uri>\n'  # noqa
+        '        <common:path>0000-0001-8135-3489</common:path>\n'
+        '        <common:host>orcid.org</common:host>\n'
+        '    </common:orcid-identifier>\n'
+        '    <person:person path="/0000-0001-8135-3489/person">\n'
+        '        <person:name visibility="public" path="0000-0001-8135-3489">\n'  # noqa
+        '            <personal-details:given-names>Lars Holm</personal-details:given-names>'  # noqa
+        '            <personal-details:family-name>Nielsen</personal-details:family-name>\n'  # noqa
+        '        </person:name>\n'
+        '        <external-identifier:external-identifiers path="/0000-0001-8135-3489/external-identifiers"/>\n'  # noqa
+        '    </person:person>\n'
+        '    <activities:activities-summary path="/0000-0001-8135-3489/activities">\n'  # noqa
+        '       <activities:employments path="/0000-0001-8135-3489/employments">\n'  # noqa
+        '           <employments:affiliation-group>\n'
+        '               <employments:employment-summary>\n'
+        '                   <employment-summary:organization>\n'
+        '                       <organization:name>CERN</organization:name>\n'
+        '                   </employment-summary:organization>\n'
+        '               </employments:employment-summary>\n'
+        '           </employments:affiliation-group>\n'
+        '       </activities:employments>\n'
+        '    </activities:activities-summary>\n'
+        '</record:record>\n'
+    )
+
+
+@pytest.fixture(scope='function')
+def names_tar_file(name_xml):
+    """Creates a Tar file with three files (two yaml) inside.
+
+    Each iteration should return the content of one yaml file,
+    it should ignore the .other file.
+    """
+    filename = Path("cli_test.tar.gz")
+    with tarfile.open(filename, "w:gz") as tar:
+        inner_filename = Path("lnielsen_name.xml")
+        with open(inner_filename, 'w') as file:
+            file.write(name_xml)
+        tar.add(inner_filename)
+        inner_filename.unlink()
+
+    yield filename
+
+    filename.unlink()  # delete created file
+
+
+def test_process(app, names_tar_file, names_service):
```

Comment:
This is a much bigger task, but we should manage to standardize CLI testing... The problem with these two approaches:
a) The current function requires an external one and does not call the corresponding click command
b) The update cmd test can only test the exit status, no read or anything because there are 2 app contexts...